docs/kubernetes/operator/instrumenting-java.md

```diff
@@ -0,0 +1,183 @@
+    - Configures the main container to use the SDK as a `java agent`.
+
+  Run `kubectl describe pod java-app-xxxx` and check:
+
+  - There should be an init container named `opentelemetry-auto-instrumentation-java` in the Pod:
+
+    ```bash
+    $ kubectl describe pod java-app-8d84c47b8-8h5z2 -n java
+    Name:             java-app-8d84c47b8-8h5z2
+    Namespace:        java
+    ...
+    ...
+    Init Containers:
+      opentelemetry-auto-instrumentation-java:
+        Container ID:  containerd://cbf67d7ca1bd62c25614b905a11e81405bed6fd215f2df21f84b90fd0279230b
+        Image:         docker.elastic.co/observability/elastic-otel-javaagent:1.0.0
+        Image ID:      docker.elastic.co/observability/elastic-otel-javaagent@sha256:28d65d04a329c8d5545ed579d6c17f0d74800b7b1c5875e75e0efd29e210566a
+        Port:          <none>
+        Host Port:     <none>
+        Command:
+          cp
+          /javaagent.jar
+          /otel-auto-instrumentation-java/javaagent.jar
+        State:          Terminated
+          Reason:       Completed
+          Exit Code:    0
+          Started:      Wed, 13 Nov 2024 15:47:02 +0100
+          Finished:     Wed, 13 Nov 2024 15:47:03 +0100
+        Ready:          True
+        Restart Count:  0
+        Environment:    <none>
+        Mounts:
+          /otel-auto-instrumentation-java from opentelemetry-auto-instrumentation-java (rw)
+          /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-swhn5 (ro)
+    ```
+
+  - The main container of the deployment is using the SDK as `javaagent`: 
+
+    ```bash
+    ...
+    Containers:
+      java-app:
+        Environment:
+    ...
+          OTEL_INSTRUMENTATION_METHODS_INCLUDE:  test.Testing[methodB]
+          JAVA_TOOL_OPTIONS:                      -javaagent:/otel-auto-instrumentation-java/javaagent.jar
+          OTEL_SERVICE_NAME:                     java-app
+          OTEL_EXPORTER_OTLP_ENDPOINT:           http://opentelemetry-kube-stack-daemon-collector.opentelemetry-operator-system.svc.cluster.local:4318
+    ...
+    ```
+
+  - The Pod has an `EmptyDir` volume named `opentelemetry-auto-instrumentation-java` mounted in both the main and the init containers in path `/otel-auto-instrumentation-java`.
+
+    ```bash
+    Init Containers:
+      opentelemetry-auto-instrumentation-java:
+    ...
+        Mounts:
+          /otel-auto-instrumentation-java from opentelemetry-auto-instrumentation-java (rw)
+    Containers:
+      java-app:
+    ...  
+        Mounts:
+          /otel-auto-instrumentation-java from opentelemetry-auto-instrumentation-java (rw)
+    ...
+    Volumes:
+    ...
+      opentelemetry-auto-instrumentation-java:
+        Type:        EmptyDir (a temporary directory that shares a pod's lifetime)
+    ```
+
+  Ensure the environment variable `OTEL_EXPORTER_OTLP_ENDPOINT` points to a valid endpoint and there's network communication between the Pod and the endpoint.
+
+4. Confirm data is flowing through in **Kibana**:
```

Comment:
There can be an intermediate step which is to confirm that the endpoint (the collector or apm server) is getting the data, but that's more detailed than might be worth putting in here

Reply:
@jackshirazi , either here or in the dedicated `troubleshooting` doc, that will definitely be important. Thanks for reminding it.

how would you verify that exactly? Are there specific logs in the DaemonSet collector (the current endpoint) about the reception and export of traces? or would you look into the application logs for that? I haven't dig yet into the logs details, but it's something we will have to address soon.

Reply:
I haven't checked yet. I think you can set the collector to output the requests to log/console, that's probably how I'd check

Reply:
That approach should better fit in a troubleshooting doc than here (making note of it for the next troubleshooting doc update). In case there's a straight forward way to determine that traces are flowing with the default installation configuration I'll add it at a later stage.
Thanks!